### ISSUE TITLE
Infinite recursion occurred in `pwndbg.gdblib.vmmap.get()` when `info proc mappings` is empty

### ISSUE THREAD

**Issue by lebr0nli** [pwndbg/pwndbg#1565]
Steps to reproduce:

```console
$ cat test.c
#include <pthread.h>
#include <stdio.h>
#include <unistd.h>
#include <stdlib.h>

void *thread_function(void *arg) {
  free(malloc(0x20));
  sleep(100);
  return NULL;
}

int main(void) {
  pthread_t thread;
  pthread_create(&thread, NULL, thread_function, NULL);
  pthread_exit(NULL);
  return 0;
}

$ gcc test.c -pthread -g -o test
$ gdb -q test -ex 'set exception-verbose on' -ex 'break thread_function' -ex 'run' -ex 'next'
```

The traceback is too long to copy, the flow which causes infinity recursion is `pwndbg.gdblib.vmmap.get()` -> `pwndbg.gdblib.vmmap.info_auxv()` -> `pwndbg.gdblib.elf.map()` -> `pwndbg.gdblib.elf.get_ehdr()` -> `pwndbg.gdblib.vmmap.find()` -> `pwndbg.gdblib.vmmap.get()`.

**Comment by gsingh93** [comment]
Seems similar to https://github.com/pwndbg/pwndbg/issues/1219

**Comment by disconnect3d** [comment]
@lebr0nli What environment do u run this on? The steps provided seems to work fine on my machine with Ubuntu 18.04

**Comment by lebr0nli** [comment]
> What environment do u run this on? The steps provided seems to work fine on my machine with Ubuntu 18.04

I tested in my Ubuntu 20.04 docker container, with GDB 11.2.

**Comment by disconnect3d** [comment]
I was able to reproduce this with `docker-compose run ubuntu20.04 bash` and then reproducing the steps above.

**Comment by disconnect3d** [comment]
This should be fixed by #1585 which also adds a regression test for this exact case.
